File: src/usuarios/Cliente.java
package usuarios;

import productos.Producto;
import tienda.Tienda;

import java.time.LocalDate;
import java.util.ArrayList;

public class Cliente {
    private String name;
    private String apellido;
    private int edad;
    private ArrayList<String> compras;

    public Cliente(String name, String apellido, int edad) {
        this.name = name;
        this.apellido = apellido;
        this.edad = edad;
        this.compras = new ArrayList<>();
    }
    public void anadirProducto(Producto producto, LocalDate fecha) {
        String detalleCompra = producto.getNombre() + ", " +
                "Precio: " + producto.getPrecio() + ", " +
                "Fecha de Importación: " + producto.getFechaImportacion() + ", " +
                "Cantidad en stock: " + producto.getStock() + ", " +
                "Número de serie: " + producto.getNumeroSerie() + ", " +
                "Fecha de compra: " + fecha;
        compras.add(detalleCompra);
    }
    public  void mostrarCompras(){
        for (int i = 0; i < compras.size(); i++) {
            String detalleCompra= compras.get(i);
            System.out.println((i+1)+"- "+detalleCompra);
        }
    }

    public String getName() {
        return name;
    }

    public String getApellido() {
        return apellido;
    }

    public int getEdad() {
        return edad;
    }

    public ArrayList<String> getCompras() {
        return compras;
    }
}

File: src/usuarios/Cliente.py

class Cliente:
    def __init__(self, name, apellido, edad):
        self.name = name
        self.apellido = apellido
        self.edad = edad
        self.compras = []

    def anadir_producto(self, producto, fecha):
        detalle_compra = f"{producto.get_nombre()}, " \
                         f"Precio: {producto.get_precio()}, " \
                         f"Fecha de Importación: {producto.get_fecha_importacion()}, " \
                         f"Cantidad en stock: {producto.get_stock()}, " \
                         f"Número de serie: {producto.get_numero_serie()}, " \
                         f"Fecha de compra: {fecha}"
        self.compras.append(detalle_compra)

    def mostrar_compras(self):
        for i, detalle_compra in enumerate(self.compras, start=1):
            print(f"{i}- {detalle_compra}")

    def get_name(self):
        return self.name

    def get_apellido(self):
        return self.apellido

    def get_edad(self):
        return self.edad

    def get_compras(self):
        return self.compras
